### ISSUE TITLE
[Question] Is there a way of setting a latency parameter for SRT?

### ISSUE THREAD

**Issue by alexsafe** [pedroSG94/RootEncoder#1673]
I saw another issue about this and I think you were saying it needs to be set on the server/streaming cloud, but I was wondering if there is a way (in the works or doable) that you know of where I can set something up app-side.

Thanks!

**Comment by pedroSG94** [comment]
Hello,

I don't set a latency value in the stream, the latency is the minimal necessary to produce and send frames.
After that, in server/player side you will have latency too. For example, a player need create a cache to work so in most of cases the latency is mainly in player side.

**Comment by alexsafe** [comment]
ok thanks. So any setting regarding latency might be set just as a parameter for example that we can attach to the srt:// . Would it even make sense to add a variable latency in the stream for the library as an improvement ?

**Comment by pedroSG94** [comment]
Hello,

Can you explain the idea? 
Do you want add a cache in seconds to the stream using that variable?

**Comment by alexsafe** [comment]
Hi,

I'm still trying to figure it out. So just food for thought for now.
It seems that the latency should be more of a server thing, the client should only request a certain value. So probably nothing to do on the library (maybe just the UrlParser to account for extra parameters). But I'm not 100% sure of this yet. The difference between buffer and latency would be smth like this(to quote from a reddit):

```
If a packet cannot be successfully retransmitted within the latency window, the packet is dropped. The buffer field is likely implemented as a helper for other protocols like TCP, where there is no fixed retransmit period.
It should help more to take any extra buffering (>200ms) and move it to the SRT latency field, where it can be used for retransmission.
```

 Rtmp uses TCP , again, from what I currently understand that means that it would benefit more from a buffer.
 SRT using UDP would benefit more from a latency setting. That would ofc need to be mirrored by the server like in this [example ]( https://callaba.io/find-the-perfect-latency-for-your-srt-setup). This [too ](https://callaba.io/find-the-perfect-latency-for-your-srt-setup)if you're interested.

**Comment by pedroSG94** [comment]
Hello,

Let me try understand your request.
SRT have a feature that re send packets if the server notify the client that it is needed and I already have this implementation.

Do you want drop packets of that feature if the buffering is higher than a custom value set to the library?
Currently that is implemented in server side, because the server notify you the packets that should be restransmitted and the library drops old packets if the server request a newer packet:
https://github.com/pedroSG94/RootEncoder/blob/master/srt/src/main/java/com/pedro/srt/srt/CommandsManager.kt#L143
So the server decide if a packet should be dropped or not. 

About the RTT using latency parameter in the url. I can check it, but I'm not sure what should I do with that value. Maybe change the TSBPD?
https://haivision.github.io/srt-rfc/draft-sharabayko-srt.html#name-timestamp-based-packet-deli

**Comment by alexsafe** [comment]
Hi Pedro,

Yes I think you are right, the only way to it is to set is as parameter for the URL. I've done a few tests (maybe way more than I should have) and I think the conclusions converge to what you said about the hardcoded value.
This is the URL that works. 
`srt://srt-auto.millicast.com:10000/?streamid=mystreamId&latency=5000000`
however when I use this with the library there is no latency, so it seems to be the default one hardcoded. I am using OBS software to benchmark and over there the same URL results in the desired delay/latency.

**Comment by pedroSG94** [comment]
Hello,

Try to compile this branch to test it:
https://github.com/pedroSG94/RootEncoder/pull/1676

**Comment by alexsafe** [comment]
Hi Pedro,

Thanks. I'm having some trouble manually adding the dependency. I followed the guide, upgraded my grade to 8.71but I got caught in a dependency error loop. This would be the latest:

```
\common\build.gradle.kts' line: 1

Error resolving plugin [id: 'com.android.library', version: '8.7.1']
> The request for this plugin could not be satisfied because the plugin is already on the classpath with an unknown version, so compatibility cannot be checked.
```

Any quick ideas on how to exit this or another way to test?

**Comment by pedroSG94** [comment]
Hello,

I compiled the commit to test it:
```gradle
  implementation 'com.github.pedroSG94.RootEncoder:library:3e5460166c'
```

**Comment by pedroSG94** [comment]
Can I assume that all is working fine? I need that info to merge the PR

**Comment by alexsafe** [comment]
hey! Sorry, I missed this completely. I just assumed you merged, for no proper reason :)
I saw you did merge it last week anyway, so I guess it will be part of 2.5.6 ?

**Comment by pedroSG94** [comment]
Yes, I will release it to the next version. Closing as completed
